lib/zhr_devs/otp/submission_check_supervisor.ex

```diff
@@ -0,0 +1,32 @@
+defmodule ZhrDevs.Otp.SubmissionSupervisor do
+  @moduledoc """
+  This is a high-level supervisor that will supervise all the submission checks.
+  """
+
+  use Supervisor
+
+  @dialyzer {:no_return, {:init, 1}}
+
+  def start_link([]) do
+    Supervisor.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @impl Supervisor
+  def init([]) do
+    children = [
+      {DynamicSupervisor, check_supervisor_opts()}
+    ]
+
+    Supervisor.init(children, strategy: :one_for_one, name: __MODULE__)
+  end
+
+  defp check_supervisor_opts do
+    [
+      strategy: :one_for_one,
+      max_children: 1,
+      max_seconds: 60,
```

Comment:
Nice, this addresses my earlier question.

Reply:
Well, no, not really. This is a strategy 3 in 60, right? But we need to also limit execution to 60-ish seconds. But also I wonder, how long would it take for us to build all the images?

Let's just make these things profusely long maybe? I don't know. 

The thing I'm thinking about now is that building scala docker files, which is the stack I'm working on these days, would probably take ten minutes because of a huge download and an unfathomably slow build time, so at the baseline perhaps we should hedge for very long runs? 

But if it works for OTMG, we can merge it like this and just make an issue.

Reply:
Let me clarify as this is something important.

Initially I was writing it before we agreed to introduce a `queue` process.
The `queue` process will be the one, who deciding what to do with each submission based on return result from running process and it's own queue of checks.

So the queue process will enforce for us that `only one submission can be running at the same time` now, instead of DynamicSupervisor.

Reply:
> The thing I'm thinking about now is that building scala docker files, which is the stack I'm working on these days, would probably take ten minutes because of a huge download and an unfathomably slow build time, so at the baseline perhaps we should hedge for very long runs?

Again, this is handled for us with the `:DOWN` message gaurantees. We should not care about it really.